.github/workflows/ci.yml

```diff
@@ -6,6 +6,12 @@ on:
   pull_request:
     branches: [ "master" ]
 
+# Configure GitHub Actions cancel in progress workflow to avoid redundant runs in pull requests.
+# See: https://docs.github.com/en/enterprise-cloud@latest/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs
+concurrency:
+  group: ${{ github.event_name == 'pull_request' && github.head_ref || 'master-group' }}
```

Comment:
How does one parse this expression? Can you add parenthesis to make the grouping clear? In particular the `master-group` in the logical expression is confusing to me...

Reply:
Now I am using conditional expression in yml for excluding master branch from canceling running CI. 
See https://docs.github.com/en/enterprise-cloud@latest/actions/writing-workflows/choosing-what-your-workflow-does/control-the-concurrency-of-workflows-and-jobs#example-only-cancel-in-progress-jobs-on-specific-branches